lib/src/messages/responses/topics/topic_subscribe.dart

```diff
@@ -18,9 +18,24 @@ class TopicSubscription implements TopicSubscribeResponse {
   String topicName;
   bool retry = true;
   final logger = Logger("TopicSubscribeResponse");
+  late Stream _broadcastStream;
 
   TopicSubscription(this._stream, this.lastSequenceNumber, this._client,
-      this.cacheName, this.topicName);
+      this.cacheName, this.topicName) {
+    _broadcastStream = _stream.asBroadcastStream();
```

Comment:
creating a broadcast stream allows for multiple subscribers to consume messages from the stream. 
In this case, there's one consumer in the init method to check if the first message is a heartbeat and the other consumer is the end user